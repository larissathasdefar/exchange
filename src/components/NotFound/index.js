import React from 'react'
import _ from 'prop-types'
import { Container, Text } from './NotFound.styles'

const NotFound = ({ history }) => {
  const redirect = () => history.push('/pockets')
  return (
    <Container>
      <Text variant="h3" onClick={redirect}>Not Found</Text>
      <Text onClick={redirect}>Click here and go back to home</Text>
    </Container>
  )
}

NotFound.propTypes = {
  user: _.shape({
    name: _.string,
    photo: _.string,
  }),
  history: _.shape({
    push: _.func,
  }).isRequired,
}

export default NotFound
