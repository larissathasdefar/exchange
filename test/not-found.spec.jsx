import React from 'react'
import { shallow } from 'enzyme'

import NotFound from 'components/NotFound'

describe('NotFound', () => {
  it('Snapshot', () => {
    const component = shallow(
      <NotFound
        history={{ push: route => console.log(route) }}
      />
    )
    expect(component.debug()).toMatchSnapshot()
  })
})