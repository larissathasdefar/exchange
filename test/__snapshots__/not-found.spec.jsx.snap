// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Snapshot 1`] = `
"<styled.div>
  <Styled(WithStyles(Typography)) variant=\\"h3\\" onClick={[Function: redirect]}>
    Not Found
  </Styled(WithStyles(Typography))>
  <Styled(WithStyles(Typography)) onClick={[Function: redirect]}>
    Click here and go back to home
  </Styled(WithStyles(Typography))>
</styled.div>"
`;
